Libraries/plist/Tools/PlistBuddy.cpp

```diff
@@ -0,0 +1,546 @@
+    fprintf(stderr, INDENT ":= <KeyPath>[:<Array Index>]      => indexes into Array\n");
+    fprintf(stderr, "\n<Type> := (string|dictionary|array|bool|real|integer|date|data)\n\n");
+#undef INDENT
+}
+
+static void
+ParseCommandKeyPathString(std::string const &keyPathString, std::queue<std::string> *keyPath) {
+    auto loc = keyPathString.find(':');
+    if (loc == std::string::npos) {
+        if (keyPathString.size() > 0) {
+            keyPath->push(keyPathString);
+        }
+    } else {
+        if (loc > 0) {
+            keyPath->push(keyPathString.substr(0, loc));
+        }
+        ParseCommandKeyPathString(keyPathString.substr(loc + 1), keyPath);
+    }
+}
+
+static std::string parseCommandValueString(std::vector<std::string>::const_iterator begin, std::vector<std::string>::const_iterator end) {
+    if (begin == end) {
+        return "";
+    }
+    std::stringstream sstream;
+    for (auto it = begin; it != end; it++) {
+        if (it != begin) {
+            sstream << ' ';
+        }
+        sstream << it->c_str();
+    }
+
+    std::string valueString = sstream.str();
+    if (valueString.front() == '\"' && valueString.back() == '\"') {
+        // strip double quote if necessary
+        valueString = valueString.substr(1, valueString.size() - 2);
+    }
+    return valueString;
+}
+
+static plist::Object *getObjectAtKeyPath(plist::Object *object, std::queue<std::string> &remainingKeys, bool leaveLastKey = true) {
+    if (remainingKeys.empty() || (leaveLastKey && remainingKeys.size() == 1)) {
+        return object;
+    } else if (!object) {
+        fprintf(stderr, "Invalid key path (indexing into null object)\n");
+        return nullptr;
+    }
+
+    std::string currentKey = remainingKeys.front();
+
+    plist::Object *subObject = nullptr;
+    switch (object->type()) {
+        case plist::ObjectType::Dictionary:
+            subObject = plist::CastTo<plist::Dictionary>(object)->value(currentKey);
+            break;
+        case plist::ObjectType::Array: {
+            char *end = NULL;
+            long long index = std::strtoll(currentKey.c_str(), &end, 0);
+            if (end == currentKey.c_str() || index < 0) {
+                fprintf(stderr, "Invalid array index\n");
+                return nullptr;
+            }
+
+            subObject = plist::CastTo<plist::Array>(object)->value(index);
+            break;
+        }
+        default:
+            // Reached a state of non-collection object with remaining key path. error
+            fprintf(stderr, "Invalid key path (indexing into non-collection object)\n");
+            return nullptr;
+    }
+
+    if (!subObject) {
+        return object;
+    } else {
+        remainingKeys.pop();
+        return getObjectAtKeyPath(subObject, remainingKeys, leaveLastKey);
+    }
+}
+
+static bool
+Print(plist::Object *object, std::queue<std::string> &keyPath, bool xml, libutil::Filesystem *fileSystem = nullptr, std::string path = "-")
```

Comment:
ProcessCommand->Print will call this without filesystem, it can be changed to call with this argument passed in if you want, but don't really see the point in changing this.